h2o-core/src/main/java/water/util/ArrayUtils.java

```diff
@@ -45,6 +45,17 @@ public static long suml(final int[] from) {
     for( int d : from ) result += d;
     return result;
   }
+  
+  public static void sum(double[][] a, double[][] b) {
+    int arrSize = a.length;
+    for (int index=0; index<arrSize; index++) {
+      a[index][index] += b[index][index];
+      for (int index2 = index+1; index2 < arrSize; index2++) {
+        a[index][index2] += b[index][index2];
+        a[index2][index] = a[index][index2];
```

Comment:
This looks weird. I would suggest to rename the function.

From sum I'd expect either a sum of all entries or pairwise sum.

By pairwise sum I mean something like:
$$\left( \begin{matrix}
1 & 2 & 3 \\
4 & 5 & 6 \\
7 & 8 & 9 
\end{matrix}\right)+\left( \begin{matrix}
1 & 0 & 0 \\
0 & 1 & 0 \\
0  & 0 & 1 
\end{matrix}\right) = \left( \begin{matrix}
2 & 2 & 3 \\
4 & 6 & 6 \\
7 & 8 & 10 
\end{matrix}\right)$$

but instead we get (if I didn't make any mistake in simulating the algo)

$$\left( \begin{matrix}
1 & 2 & 3 \\
4 & 5 & 6 \\
7 & 8 & 9 
\end{matrix}\right)+\left( \begin{matrix}
1 & 0 & 0 \\
0 & 1 & 0 \\
0  & 0 & 1 
\end{matrix}\right) = \left( \begin{matrix}
2 & 2 & 3 \\
2 & 6 & 6 \\
3 & 6 & 10 
\end{matrix}\right)$$

Reply:
It is for elementwise addition of two symmetric matrices.  I changed the name.